ksqldb-tools/src/main/java/io/confluent/ksql/tools/migrations/util/CommandParser.java

```diff
@@ -214,8 +221,35 @@ private static SqlCommand transformToSqlCommand(final String sql) {
             parsedStatement.getValues(),
             parsedStatement.getColumns().stream()
                 .map(ColumnName::text).collect(Collectors.toList()));
-      case CONNECTOR:
-        return new SqlConnectorStatement(sql);
+      case CREATE_CONNECTOR:
+        final CreateConnector createConnector;
+        try {
+          createConnector = (CreateConnector) new AstBuilder(TypeRegistry.EMPTY)
+              .buildStatement(KSQL_PARSER.parse(sql).get(0).getStatement());
+        } catch (ParseFailedException e) {
+          throw new MigrationException(String.format(
+              "Failed to parse CREATE CONNECTOR statement. Statement: %s. Reason: %s",
+              sql, e.getMessage()));
+        }
+        return new SqlCreateConnectorStatement(
+            sql,
+            createConnector.getName(),
+            createConnector.getType() == Type.SOURCE,
+            createConnector.getConfig().entrySet().stream()
+                .collect(Collectors.toMap(e -> e.getKey(), e -> toFieldType(e.getValue())))
+        );
+      case DROP_CONNECTOR:
+        final Matcher dropConnectorMatcher = DROP_CONNECTOR.matcher(sql);
+        if (!dropConnectorMatcher.matches()) {
+          throw new MigrationException("Invalid DROP CONNECTOR command: " + sql);
+        }
+        final String connectorName = dropConnectorMatcher.group(1);
+        return new SqlDropConnectorStatement(
+            sql,
+            connectorName.startsWith("`")
```

Comment:
See my comment about case-sensitivity in the Java client on your other PR. Also, even if we were to go that route, this logic here is not correct since case-sensitivity of identifiers can also be preserved with double quotes (not just backticks).

Reply:
Updated to pass quotes to the java client. I also added quotes to the SqlCreateConnector name, since the parser removes quotes. Is this something we would need to do for `INSERT INTO...VALUES` as well? I'm not sure how the endpoint deals with case-sensitive names.

Reply:
False alarm - backquotes are added to the INSERT INTO source names at another point. I added a refactor to this commit so that the quotes are added in `CommandParser`, since the logic is used twice

Reply:
Thanks for checking. What about the INSERT VALUES field names? Those should have the same behavior as the stream name.

Reply:
See comment below on why `preserveCase` is public